tests/clients/test_ngsi_ld_query.py

```diff
@@ -0,0 +1,311 @@
+            None
+        """
+        #Extra size parameters for modular testing
+        self.cars_nb = 500
+        self.span = 3
+        
+        #client parameters
+        self.fiware_header = FiwareLDHeader(ngsild_tenant=settings.FIWARE_SERVICE)
+        self.cb = ContextBrokerLDClient(fiware_header=self.fiware_header,
+                                        url=settings.LD_CB_URL)
+
+        #Prep db
+        try:
+            entity_list = True
+            while entity_list:
+                entity_list = self.cb.get_entity_list(limit=1000)
+                self.cb.entity_batch_operation(action_type=ActionTypeLD.DELETE,
+                                                   entities=entity_list)
+        except RequestException:
+            pass
+        
+        #base id
+        self.base='urn:ngsi-ld:'
+        
+        #Some entities for relationships
+        self.garage = ContextLDEntity(id=f"{self.base}garage0",type=f"{self.base}gar")
+        self.cam = ContextLDEntity(id=f"{self.base}cam0",type=f"{self.base}cam")
+        
+        #Entities to post/test on
+        self.cars = [ContextLDEntity(id=f"{self.base}car0{i}",type=f"{self.base}car") for i in range(0,self.cars_nb-1)]
+        
+        #Some dictionaries for randomizing properties
+        self.brands = ["Batmobile","DeLorean","Knight 2000"]
+        self.addresses = [
+            {
+                "country": "Germany",
+                "street-address": {
+                    "street":"Mathieustr.",
+                    "number":10},
+                "postal-code": 52072
+            },
+            {
+                "country": "USA",
+                "street-address": {
+                    "street":"Goosetown Drive",
+                    "number":810},
+                "postal-code": 27320
+            },
+            {
+                "country": "Nigeria",
+                "street-address": {
+                    "street":"Mustapha Street",
+                    "number":46},
+                "postal-code": 65931
+            },
+        ]
+
+        #base properties/relationships
+        self.humidity = NamedContextProperty(name="humidity",value=1)
+        self.temperature = NamedContextProperty(name="temperature",value=0);
+        self.isParked = NamedContextRelationship(name="isParked",object="placeholder")
+        self.isMonitoredBy = NamedContextRelationship(name="isMonitoredBy",object="placeholder")
+        
+        #q Expressions to test
+        self.qs = [
+            'temperature > 0',
+            'brand != "Batmobile"',
+            'isParked | isMonitoredBy',
+            'isParked == "urn:ngsi-ld:garage0"',
+            'temperature < 60; isParked == "urn:ngsi-ld:garage0"',
+            '(temperature >= 59 | humidity < 3); brand == "DeLorean"',
+            '(isMonitoredBy; temperature<30) | isParked',
+            '(temperature > 30; temperature < 90)| humidity <= 5',
+            'temperature.observedAt >= "2020-12-24T12:00:00Z"',
```

Comment:
It seems like the `observedAt` is not generated automatically. @SystemsPurge Can you add exactly this timestamp `"2020-12-24T12:00:00Z"` to the `temperature` of some car entities? And then we can see whether this condition work